src/plugins/preauth/pkinit/pkinit_crypto_openssl.c

```diff
@@ -2970,24 +2975,27 @@ server_process_dh(krb5_context context,
     pub_key = BN_to_ASN1_INTEGER(server_pubkey, NULL);
     if (pub_key == NULL)
         goto cleanup;
-    *dh_pubkey_len = i2d_ASN1_INTEGER(pub_key, NULL);
-    if ((p = *dh_pubkey = malloc(*dh_pubkey_len)) == NULL)
+    dh_pubkey_len = i2d_ASN1_INTEGER(pub_key, NULL);
+    if ((p = dh_pubkey = malloc(dh_pubkey_len)) == NULL)
```

Comment:
While you're touching this line, please move the function call outside of the `if`'s conditional.